### PR TITLE
Les - 282 PhoneNumberPage 

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

**Review by valer-sky — APPROVED** [review]

File: src/components/PhoneNumber.tsx [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
-      <input
-        className='phoneNumberInput'
-        type='text'
-        onChange={fieldHandler}
-        value={value}
-        required
-      />
+      <label className='phoneNumberLabel'>Phone number
+        <PhoneInput
+          onChange={(value: string, country: any, e: React.ChangeEvent<HTMLInputElement>, formattedValue: string) => {
```
Comment (Terapool) [inline src/components/PhoneNumber.tsx:19]: the function is long, declare it separately

File: src/components/PhoneNumber.tsx [hunk 2]
```diff
@@ -1,27 +1,44 @@
 import {useState} from "react";
 import "./phoneNumber.scss";
+import PhoneInput from 'react-phone-input-2';
```
Comment (Terapool) [inline src/components/PhoneNumber.tsx:3]: react-phone-input-2 - there is no such module./

You need to add this npm dependancy in package.json to this pull request

File: src/App.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ function App(): any {
 
   return (
     <BrowserRouter>
-      <FirstRegistrationForm />
+      <PhoneNumberPage/>
     </BrowserRouter>
```
Comment (Evgeniy121212) [inline src/App.tsx:15]: the App.tsx was changed - upload the new version from develop

File: src/pages/PhoneNumberPage.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import "../components/modal/modal.scss";
+import Button from "../components/Button";
+import PhoneNumber from "../components/PhoneNumber";
+import {useState} from "react";
+
+const PhoneNumberPage = () => {
+  const [error, setError] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('375');
```
Comment (Evgeniy121212) [inline src/pages/PhoneNumberPage.tsx:8]: move '375' to constant to start of that file

File: src/pages/PhoneNumberPage.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div className='field'>
+      <div className='modal'>
+        <h2 className='title'>Enter your phone number</h2>
+        <PhoneNumber setError={setError}
+                     error={error}
+                     phoneNumber={phoneNumber}
+                     setPhoneNumber={setPhoneNumber}/>
+        <Button buttonType={'submit'}
+                buttonText={'Submit code'}/>
```
Comment (Evgeniy121212) [inline src/pages/PhoneNumberPage.tsx:19]: don't use that brackets `{}` if it's text variable; just put string like `buttonType="submit"`

File: src/components/PhoneNumber.tsx (hunk 2 above)
Comment (Terapool) [reply src/components/PhoneNumber.tsx:3]: Dıd you fıxed thıs comment?

Comment (Kurilo-Pavel) [reply src/components/PhoneNumber.tsx:3]: "react-datepicker": "^4.8.0",
    "react-dom": "^18.2.0",
    "react-feather": "^2.0.10",
    "react-phone-input-2": "^2.15.1",
    "react-redux": "^8.0.4",
    "react-router-dom": "^6.4.2",
    "react-scripts": "5.0.1",

File: src/App.tsx [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
   return (
     <BrowserRouter>
-      <FirstRegistrationForm />
+      <div className="App">
+        <Body />
+        <Routes>
+          <Route path='/users/sign_in' element={ <PhoneNumberPage/>}/>
+          <Route path='/users/sign_up' element={<FirstRegistrationForm/>}/>
+        </Routes>
+      </div>
```
Comment (Evgeniy121212) [inline src/App.tsx:16]: Please, resolve conflicts into `App.tsx` file with according on current `develop`-branch state

File: src/pages/PhoneNumberPage.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import "../components/modal/modal.scss";
+import Button from "../components/Button";
+import PhoneNumber from "../components/PhoneNumber";
+import {useState} from "react";
+
+const defaultCauntry = '375';
```
Comment (Evgeniy121212) [inline src/pages/PhoneNumberPage.tsx:6]: typo: "defaultCauntry" -> "defaultCountry"

**Review by OxygenO22 — APPROVED** [review]

**Review by Terapool — APPROVED** [review]

**Pull request merged**